### PR TITLE
feat(docs-page): strip tfe version from pathparams

### PR DESCRIPTION
**Pull request by thiskevinwang**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

This fixes a bug where a frontend URL was being converted to an incorrect content API url
- /enterprise/v202206-1/support
- https://content.hashicorp.com/api/content/ptfe-releases/doc/v2022.6.1/enterprise/v202206-1/support

```console
[GET] /enterprise/v202206-1/support
10:36:18:02
2022-06-30T14:36:18.253Z	e224ee77-631b-4fc5-a2ce-f47460eec718	ERROR	Failed to generate static props: ContentApiError: Failed to fetch: https://content.hashicorp.com/api/content/ptfe-releases/doc/v2022.6.1/enterprise/v202206-1/support
    at fetchDocument (/var/task/.next/server/chunks/935.js:12159:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 0)
    at async RemoteContentLoader.loadStaticProps (/var/task/.next/server/chunks/935.js:12767:41)
    at async getStaticProps (/var/task/.next/server/chunks/935.js:12849:31)
    at async getStaticProps (/var/task/.next/server/pages/enterprise/[[...page]].js:84:17)
    at async Object.renderToHTML (/var/task/node_modules/next/dist/server/render.js:470:20)
    at async doRender (/var/task/node_modules/next/dist/server/base-server.js:867:38)
    at async cacheEntry.responseCache.get.isManualRevalidate.isManualRevalidate (/var/task/node_modules/next/dist/server/base-server.js:966:28)
    at async /var/task/node_modules/next/dist/server/response-cache.js:72:36 {
  status: 404
```

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Review by zchsh — APPROVED** [review]
This looks good to me!

Full transparency: I'm definitely leaning on the presence of tests for `stripVersionFromPathParams` in this review. The revised `REGEX` and `TFE_REGEX` both make sense to me, but can't say I understand the previous `REGEX` very well 😬

File: packages/docs-page/util/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-const REGEX =
-  /^v([0-9]+)\.([0-9]+)\.(x|[0-9]+)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?$/i
```
Comment (BRKalow) [inline packages/docs-page/util/index.ts:2]: Are we potentially losing anything by simplifying this regex? 💭

Comment (thiskevinwang) [reply packages/docs-page/util/index.ts:2]: @zchsh @BRKalow the `REGEX` before this PR was potentially an oversight. It'd match semvers w/ RC suffixes, like `v1.2.3-rc.1`.

The `-rc.1` suffix is what we'd lose by simplifying the regex

Comment (BRKalow) [reply packages/docs-page/util/index.ts:2]: Makes sense! Just a future thought, if/when we support pre-releases we might have to bring this back (no action needed now)

**Review by BRKalow — APPROVED** [review]

**Pull request merged**